modules/selenium/src/main/java/org/testcontainers/containers/RecordingFileFactory.java

```diff
@@ -1,15 +1,16 @@
 package org.testcontainers.containers;
 
 import org.junit.runner.Description;
+import org.testcontainers.containers.VncRecordingContainer.VncRecordingFormat;
 
 import java.io.File;
 
 public interface RecordingFileFactory {
 
     @Deprecated
-    default File recordingFileForTest(File vncRecordingDirectory, Description description, boolean succeeded) {
-        return recordingFileForTest(vncRecordingDirectory, description.getTestClass().getSimpleName() + "-" + description.getMethodName(), succeeded);
+    default File recordingFileForTest(File vncRecordingDirectory, Description description, boolean succeeded, VncRecordingFormat recordingFormat) {
+        return recordingFileForTest(vncRecordingDirectory, description.getTestClass().getSimpleName() + "-" + description.getMethodName(), succeeded, recordingFormat);
     }
 
-    File recordingFileForTest(File vncRecordingDirectory, String prefix, boolean succeeded);
+    File recordingFileForTest(File vncRecordingDirectory, String prefix, boolean succeeded, VncRecordingFormat recordingFormat);
```

Comment:
This breaks existing implementations, I'm not sure what the most elegant solution would be.

This would create a stackoverflow if you don't implement any of the new methods, but if you already have one that is using the three param `recordingFileForTest` it would work. 
```java
@Deprecated
default File recordingFileForTest(File vncRecordingDirectory, String prefix, boolean succeeded) { 
    return recordingFileForTest(vncRecordingDirectory, prefix, boolean, null);
}
    default File recordingFileForTest(File vncRecordingDirectory, String prefix, boolean succeeded, VncRecordingFormat recordingFormat) {
    return recordingFileForTest(vncRecordingDirectory, prefix, boolean);
}
```
The alternative would be to just `return null` from the three param default method, since our code would call the four param version. In any case we would need to add default methods so that users are not forced to implement an unused method.

Reply:
Indeed, I used `Default` for the new method, so user can implement it only if `mp4` is desired, or setting `flv` explicitly.
I canceled the changes in the three params method, I didn't use `Deprecated` and I considered It as a legitimate method (recording with default format).

Thanks.